File: Find First and Last Position of Element in Sorted Array Leet.py
# Medium
import numpy as np

arr = np.zeros(6, dtype=np.int8)

target = int(input('Enter target value: '))

flag = 0

result_arr = np.zeros(2, dtype=np.int8)

for x in range(0, len(arr)):
    arr[x] = int(input(f"Enter element {x + 1}: "))

for i in range(0, len(arr)):
    if flag == 0:
        if arr[i] == target:
            result_arr[0] = i
            flag = 1
    
    if i == 5 and flag == 0:
        result_arr[0], result_arr[1] = -1, -1

    else:
        if arr[i] == target:
            result_arr[1] = i

print(result_arr)

File: FindFirstandLastPositionofElementinSortedArrayLeet.cpp
#include<iostream>
#include<stdlib.h>
using namespace std;
int main()
{
    system("cls");
    int size, target, flag = 0, arr[100], res[2], find = 0;
    cout<<"Enter the size of the array: ";
    cin>>size;
    cout<<"Enter array: "<<endl;
    for(int i=0;i<size;i++)
    {
        cin>>arr[i];
    }
    cout<<"Enter target: ";
    cin>>target;
    for(int i=0;i<size;i++)
    {
        if(arr[i] == target)
        {
            if(flag == 0)
            {
                int fpos = i;
                res[0] = fpos;
                flag = 1;
                find = 1;
            }
            else
            {
                int lpos = i;
                res[1] = lpos;
                find = 2;   
            }
        }
    }
    if(find == 2)
    {
        for(int i=0;i<2;i++)
        {
            cout<<res[i]<<", ";
        }
    }
    else
    {
        cout<<"Either one or both instances of target were missing from the array"<<endl;
    }   
    return 0;
}